### ISSUE TITLE
 redirecting page when hostname is not right

### ISSUE THREAD

**Issue by laysdra7265** [hackmdio/codimd#630]
I deploymented own-hosting server for china companion.
I had to turn off cdn,because the foreign cdn resources is extremely slow with some china ISP.
We have `example.com` and `www.example.com` can access our hackmd,but have some CORS policy problem with font resource.
Even though I add `example.com` to alloworigin,the browser still not allow `example.com` to load font resource from `www.example.com`.
Shouldn't we do something like this to make alloworigin effective on all request?
```js
app.all('/', function (req, res, next) {
  var host = req.get('host')
  if (config.alloworigin.indexOf(host) !== -1) {
    res.header('Access-Control-Allow-Origin', '*')
  }
  next()
})
```
.

**Comment by SISheogorath** [comment]
Can you share the browser error message? It would help to fix the problem more generic.

CORS is an important security feature which should be handled correctly. And providing no policy when the host is not part of the policy is totally pointless since it's breaking the intention behind CORS.

Let's do it correct 😄  So when you can provide the error message, let's see what we can do.

**Comment by laysdra7265** [comment]
Something like `principle of least privilege`,right?
Then how about this?I know the importance of the security feature.
```js
app.all('/', function (req, res, next) {
  var host = req.get('host')
  if (config.alloworigin.indexOf(host) !== -1) {
    res.header('Access-Control-Allow-Origin',host)
  }
  next()
})
```
There is the error logs from chrome console.

```console
Access to Font at 'http://www.example.com:3000/build/af7ae505a9eed503f8b8e6982036873e.woff2' from origin 'http://example.com:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
Access to Font at 'http://www.example.com:3000/fonts/SourceSansPro-Regular.woff' from origin 'http://example.com:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
Access to Font at 'http://www.example.com:3000/fonts/SourceSansPro-Semibold.woff' from origin 'http://example.com:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
Access to Font at 'http://www.example.com:3000/build/fee66e712a8a08eef5805a46892932ad.woff' from origin 'http://example.com:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
Access to Font at 'http://www.example.com:3000/build/b06871f281fee6b241d60582ae9369b9.ttf' from origin 'http://example.com:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
Failed to load http://www.example.com:3000/me: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
Access to Font at 'http://www.example.com:3000/fonts/SourceSansPro-Italic.woff' from origin 'http://example.com:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://example.com:3000' is therefore not allowed access.
```

**Comment by laysdra7265** [comment]
Making a page redirecting when hostname is not right like the case I encounter maybe a better choice?
example:
example.com -> www.example.com

**Comment by SISheogorath** [comment]
Yes, redirecting would be the better choice.

When you set up your service on a different URL than the URL created by the config object, it should redirect you there. This will break all not-correct setups, but with some documentation about it, it should be fine and people should make it improve their installations 👍

**Comment by ccoenen** [comment]
if the enhancement is meant for the redirecting part: Please also change the title of the issue.

**Comment by SISheogorath** [comment]
I just decided that it's not our job to redirect this.

Reason is simple: We recommend to use a reverse proxy in front of CodiMD in **all** configurations.

We also only listen on one port, which means we don not redirect from http to https. So further more, you need anyway a service in front of CodiMD to run it in a useful way on HTTPS (which we consider as the default and wanted case for user facing services) and this reverse proxy can and should do the redirect.
